packages/camera/camera/test/camera_test.dart

```diff
@@ -32,9 +68,443 @@ void main() {
+        () async {
+      CameraController cameraController = CameraController(
+          CameraDescription(
+              name: 'cam',
+              lensDirection: CameraLensDirection.back,
+              sensorOrientation: 90),
+          ResolutionPreset.max);
+      await cameraController.initialize();
+
+      mockPlatformException = true;
+      expect(
+          cameraController.takePicture(),
+          throwsA(isA<CameraException>().having(
+            (error) => error.description,
+            'foo',
+            'bar',
+          )));
+      mockPlatformException = false;
+    });
+
+    test('startImageStream() throws $CameraException when uninitialized', () {
+      CameraController cameraController = CameraController(
+          CameraDescription(
+              name: 'cam',
+              lensDirection: CameraLensDirection.back,
+              sensorOrientation: 90),
+          ResolutionPreset.max);
+      expect(
+          cameraController.startImageStream((image) => null),
+          throwsA(isA<CameraException>().having(
+            (error) => error.description,
+            'Uninitialized CameraController.',
+            'startImageStream was called on uninitialized CameraController.',
+          )));
+    });
+
+    test('startImageStream() throws $CameraException when recording videos',
+        () async {
+      CameraController cameraController = CameraController(
+          CameraDescription(
+              name: 'cam',
+              lensDirection: CameraLensDirection.back,
+              sensorOrientation: 90),
+          ResolutionPreset.max);
+      await cameraController.initialize();
+
+      cameraController.value =
+          cameraController.value.copyWith(isRecordingVideo: true);
+      expect(
+          cameraController.startImageStream((image) => null),
+          throwsA(isA<CameraException>().having(
+            (error) => error.description,
+            'A video recording is already started.',
+            'startImageStream was called while a video is being recorded.',
+          )));
+    });
+    test(
+        'startImageStream() throws $CameraException when already streaming images',
+        () async {
+      CameraController cameraController = CameraController(
+          CameraDescription(
+              name: 'cam',
+              lensDirection: CameraLensDirection.back,
+              sensorOrientation: 90),
+          ResolutionPreset.max);
+      await cameraController.initialize();
+
+      cameraController.value =
+          cameraController.value.copyWith(isStreamingImages: true);
+      expect(
+          cameraController.startImageStream((image) => null),
+          throwsA(isA<CameraException>().having(
+            (error) => error.description,
+            'A camera has started streaming images.',
+            'startImageStream was called while a camera was streaming images.',
+          )));
+    });
+
+    test('startImageStream() calls CameraPlatform', () async {
```

Comment:
Would it make sense to split the start/stop ImageStream tests to their own files, so we don't mix the PlatformInterface-based initialization and tests, with the MethodChannel ones?

Reply:
(Also for later cleanup, it's easier, just deleting a file)

Reply:
I agree, thanks for pointing it out. I moved the tests into a new file called `camera_image_stream_test.dart`.